src/fixupReactRouter.js

```diff
@@ -0,0 +1,52 @@
+
+function ensureArray(maybeArray) {
+  if (!maybeArray) {
+    return [];
+  } else {
+    if (Array.isArray(maybeArray)) {
+      return maybeArray;
+    } else {
+      return [maybeArray];
+    }
+  }
+}
+
+function isReactRouterish(type) {
+  return type && (
+    type.displayName === 'Router' ||
+    type.name === 'Router' // In case Ryan and Michael embrace ES6 classes
+  );
+}
+
+function componentsToForceUpdate(routes) {
+  return routes
+    .map(route => {
+      // The route properties are at different locations depending on if plain routes
+      // or regular element routes are used
+      const component = route.component || (route.props && route.props.component);
+      const componentsMap = route.components || (route.props && route.props.components) || {};
+      const childRoutes = route.childRoutes || (route.props && ensureArray(route.props.children));
+
+      const components = Object.keys(componentsMap).map(key => componentsMap[key]);
+      const childRouteComponents = childRoutes && componentsToForceUpdate(childRoutes);
+
+      return [].concat(component, components, childRouteComponents);
+    })
+    .reduce((reduceTarget, components) => reduceTarget.concat(components), []) // flatten
+    .filter(c => typeof(c) === 'function');
+}
+
+function fixupReactRouter(props, forceUpdateFunction) {
+  // Ideally we want to teach React Router to receive children.
+  // We're not in a perfect world, and a dirty workaround works for now.
+  // https://github.com/reactjs/react-router/issues/2182
+  const routes = props.routes || ensureArray(props.children);
+  const components = componentsToForceUpdate(routes);
+  for (const component of components) {
```

Comment:
[`for of` requires `Symbol`s](http://babeljs.io/repl/#?evaluate=true&presets=es2015%2Creact&experimental=false&loose=false&spec=false&code=for%20(var%20x%20of%20y)%20%7B%0A%20%20%0A%7D), let’s use `components.forEach`